### ISSUE TITLE
RFE: Easy way to query for supported rpmlib features

### ISSUE THREAD

**Issue by ignatenkobrain** [rpm-software-management/rpm#717]
I could not find way how to query rpm whether it supports some rpmlib feature. This would be very useful for tools which want to check whether rich deps or dynamic builddeps are supported.

**Comment by ffesti** [comment]
This is something we really should add. The right way to do this is probably to have a macro per feature in the spec file syntax. I realized we don't even have a macro with the rpm version - which is kinda strange.
So we need a naming scheme for these macros. Some of these can be derived from rpmlib() Provides. But we also need some new ones that are spec file only and some of he rpmlib() Provides are not really useful as they are install only - like payload compression.

**Comment by ffesti** [comment]
Btw: rpm.ds.Rpmlib() in Python and rpmdsRpmlib() in C do provide the rpmlib features supported.

**Comment by pmatilai** [comment]
Yup. The thing is that rpmlib() features track installability of produced packages, because you don't know what rpm version will be used to install the procuced package, and because of rpmlib() deps, you don't need to. Build-time features tracking is quite a different, but I agree, useful thing.

**Comment by ffesti** [comment]
Managing build features is a pretty big topic. Tools using rpmbuild are probably not even the most interesting use case. Spec files will also be interested in what features they can use. That would allow spec to be compatible over a wide range of rpm(build) versions.
Unfortunately rpm does not track those things at all right now.

**Comment by pmatilai** [comment]
On a related note, in rpm >= 4.17 you'll be able to test for supported built-in macros directly with the macro language itself, eg `rpm --eval "%{defined:macrobody}'`
